### PR TITLE
MAT-2686

### PR DESCRIPTION
**Pull request by danielschweigert-TRI**
Changes to biologic data file parsing to accommodate new file layout.

### CODE REVIEW

**Review by moorepatrick — APPROVED** [review]
LGTM

File: beep/conversion_schemas/biologic_conversion.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ data_columns:
   cycle number:
```
Comment (patrickherring-TRI) [inline beep/conversion_schemas/biologic_conversion.yaml:6]: @willpowelson @moorepatrick Is the cycle number where the cycle index being stored?

Comment (willpowelson) [reply beep/conversion_schemas/biologic_conversion.yaml:6]: no, cycle index is stored under the header `cycle_index` which we add to the files in the pipeline.

Cycle number is calculated by Biologic based on charge/discharge alternance and is not what we want.

Comment (willpowelson) [reply beep/conversion_schemas/biologic_conversion.yaml:6]: see https://github.com/TRI-AMDD/beep-sync/pull/17#issuecomment-870810296
for the discussion where we debugged this in BEEP-sync

Comment (patrickherring-TRI) [inline beep/conversion_schemas/biologic_conversion.yaml:6]: @danielschweigert-TRI Can you change this to cycle_index?

**Review by patrickherring-TRI — CHANGES_REQUESTED** [review]
I think that cycle_index should be a column already in the data, could you use that value instead?

File: beep/conversion_schemas/biologic_conversion.yaml (hunk 1 above)
Comment (danielschweigert-TRI) [reply beep/conversion_schemas/biologic_conversion.yaml:6]: @patrickherring-TRI I don't see a `cycle_index` among the column names. The column names header is:
```
mode;ox;error;time/s;Ecell/V;dq/mA.h;I/mA;(Q-Qo)/mA.h;x;control/V;control/mA;control changes;dQ/mA.h;cycle number;DQ/mA.h;Rapp/Ohm;control/°C;T/°C;rotation rate/rpm;Ns changes;freq/Hz;|Ewe|/V;|I|/A;Phase(Z)/deg;|Z|/Ohm;Re(Z)/Ohm;-Im(Z)/Ohm;I Range;Q charge/mA.h;Q discharge/mA.h;Tech Num;End buf;counter inc.;R/Ohm;P/W;Energy/W.h;Analog OUT/V;Cs-2/µF-2;Re(Y)/Ohm-1;Im(Y)/Ohm-1;|Y|/Ohm-1;Phase(Y)/deg;Energy charge/W.h;Energy discharge/W.h;Capacitance charge/µF;Capacitance discharge/µF;Ns;dI/dt/mA/s;Delta(mass)/g;Phase(I)/rad;Rcmp/Ohm;Cs/µF;sin ampl/V;Conductivity/S.cm-1;Cp/µF;Cp-2/µF-2;Efficiency/%;(Q-Qo)/C;dQ/C;cycle time/s;charge time/s;discharge time/s;d(Q-Qo)/dE/mA.h/V;Capacity/mA.h;<EcellX>/V;Temperature/°C/°c;z cycle;NbIterInstr;Instr;Loop
```

**Comment by patrickherring-TRI** [comment]
You might have the wrong file... @moorepatrick did you send the file resulting from concatenation?

File: beep/conversion_schemas/biologic_conversion.yaml (hunk 1 above)
Comment (patrickherring-TRI) [reply beep/conversion_schemas/biologic_conversion.yaml:6]: Those look like the column names pre concatenation... I can look at this later and regenerate the file if necessary

Comment (danielschweigert-TRI) [reply beep/conversion_schemas/biologic_conversion.yaml:6]: My fault; I picked the wrong one which is similarly named and one of the pre-concatenated ones.

However, the correct file file does not appear to have a corresponding meta-data file (with `.mpl` extension) in the bucket.

Comment (patrickherring-TRI) [reply beep/conversion_schemas/biologic_conversion.yaml:6]: It should be the same metadata file as the other file... You can just rename the metadata file to match and we will fix that in the syncer

**Review by patrickherring-TRI — APPROVED** [review]
LGTM!

**Pull request merged**